functions/controllers/retrieveM2MToken/index.js

```diff
@@ -0,0 +1,30 @@
+const functions = require("firebase-functions");
+const fetch = require("node-fetch");
+const { db } = require("../../utils/admin");
+
+const getM2MToken = () => {
+  options = {
+    method: "POST",
+    headers: {
+      "Content-Type": "application/json",
+    },
+    body: JSON.stringify({
+      client_id: functions.config().client.client_id,
+      client_secret: functions.config().client.client_secret,
+      audience: "https://cruzhacks.us.auth0.com/api/v2/",
```

Comment:
You have to be really careful about storing URLs. We want to make sure we don't leak any environment we are making requests to and this endpoint will vary depending on the environment. Check out [line 54](https://github.com/CruzHacks/cruzhacks-2022-backend/pull/18/files) of index.js as reference. It uses the same domain.